src/io/sampling.jl

```diff
@@ -284,7 +284,9 @@ function _deactivate_interventions(to_update::DataFrame)::Nothing
         _row = to_update.fieldname .== c
         _bnds = length(to_update[_row, :bounds][1]) == 2 ? (0.0, 0.0) : (0.0, 0.0, 0.0)
 
-        dval = to_update[_row, :ptype][1] == "integer" ? 0 : 0.0
+        is_int = to_update[_row, :ptype][1] == "integer"
+        is_cat = to_update[_row, :ptype][1] == "categorical"
+        dval = (is_int .| is_cat) ? 0 : 0.0
```

Comment:
```suggestion
        # Handle special cases for discrete valued factors.
        # This will break if meanings/possible values of `ptype` changes.
        is_int = to_update[_row, :ptype][1] == "integer"
        is_cat = to_update[_row, :ptype][1] == "categorical"
        dval = (is_int .| is_cat) ? 0 : 0.0
```